### PR TITLE
Refactor settings

### PR DESCRIPTION
**Pull request by matti-lamppu**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Refactors how the project's settings are defined
    - Use [django-environment-config](https://github.com/MrThearMan/django-environment-config) for configuring how different environments are set up (at first, I tried using [django-configurations](https://github.com/jazzband/django-configurations), but I wasn't too happy with how [magical it was](https://github.com/jazzband/django-configurations/issues/360), [so I decided to write my own](https://xkcd.com/927/))
- Some small refactorings

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of the fastest way to test your changes."

- Automated tests
- Backend devs should check that it works locally following the setup
- Frontend devs should check that the docker setup works for them

## 🚧 Dependencies
[//]: # "Is this PR dependent on other changes outside this repository? Describe the changes, or add links to them."
[//]: # "e.g. This PR breaks X and is waiting for frontend changes before merging."

- None

### CODE REVIEW

**Comment by matti-lamppu** [comment]
This requires testing in Platta to check that all environment variables are in there. Note that all environments should be checked, or at least their variables.

**Review by ranta — CHANGES_REQUESTED** [review]
Really nice work 🚀 

Discussed some issues off-platform, which should be addressed before merging

File: README.md [hunk 1]
```diff
@@ -233,31 +271,12 @@ The package has integrations in all the relevant parts of the project
...
-to it in order to enable the debug toolbar:
-
-```python
-try:
-    # Assure debug toolbars are installed
-    import debug_toolbar
-    import graphiql_debug_toolbar
+You should add a `local_settings.py` on the root level of the project and add
+a two classes called `LocalMixin` and `AutomatedTestMixin` to it.
+These can be used to override settings for local development and automated tests respectively.
```
[the remark this answers is not in the document]
Comment (matti-lamppu) [reply README.md]: Fixed.

**Review by ranta — APPROVED** [review]
# Magical 🧙🪄

**Pull request merged**